src/components/navigation/navigationItem.svelte

```diff
@@ -0,0 +1,150 @@
+<script lang="ts">
+  import { createEventDispatcher, onMount } from 'svelte'
+  import type { SvelteHTMLElements } from 'svelte/elements'
+  import type { IconName } from '../../../icons/meta'
+  import Icon from '../icon/icon.svelte'
+
+  // This black magic comes from this thread:
+  // https://github.com/sveltejs/language-tools/issues/442#issuecomment-1278618531
+  //
+  // To quote that thread - This is "absolute bonkers!"
+  //
+  // It's interesting, minor variations which I would expect to work on don't,
+  // and this is the only combination which seems to do what we want and I'm not
+  // clear on why. You're welcome to try other approaches here.
+  //
+  // Tips, for if things aren't working right:
+  // 1) npm run gen-types
+  // 2) Reload your VSCode Window (sometimes the Svelte Type Checker struggles).
+  // 3) Make sure any script tags on your component have a `lang="ts"` attribute.
+  type Href = $$Generic<string | undefined>
+  type Disabled = $$Generic<undefined extends Href ? boolean : undefined>
+  type ExcludedProps = 'size' | 'href' | 'hreflang'
+
+  type CommonProps = {
+    inList?: boolean
+    icon?: IconName
+  }
+
+  type ButtonProps = CommonProps &
+    Omit<Partial<SvelteHTMLElements['button']>, ExcludedProps> & {
+      isDisabled?: Disabled
+      isLoading?: boolean
+      href?: never
+    }
+
+  type LinkProps = CommonProps &
+    Omit<Partial<SvelteHTMLElements['a']>, ExcludedProps> & {
+      href: Href
+      isCurrent?: boolean
+    }
+
+  type $$Props = LinkProps | ButtonProps
+
+  export let href: Href = undefined
+  export let icon: IconName = undefined
+  export let isLoading: boolean = false
+  export let isDisabled: boolean = false
+  export let isCurrent: boolean = window.location.pathname === href
+  export let inList: boolean = true
+
+  const checkIfCurrent = () => {
+    isCurrent = window.location.pathname === href
+  }
+
+  $: tag = href ? 'a' : ('button' as 'a' | 'button')
```

Comment:
This is inexplicably broken when using with web components.

Reply:
Broken because custom elements doesn't seem to like the double `<svelte:elements>` bit.

Reply:
Svelte is a bit funky :/

Reply:
Hmm okay. It does make me feel a little weird landing it without working WC support, as a lot of the usages for these components are in Brave Core, which is solely WC based. 

That said, I want to unblock you, so I guess we can land as is, if you're okay to come back and try and fix it when you have some time?

I'd prefer to avoid a situation where we have a bunch of functionality which only works in Svelte, and a bunch which only works in WCs.

Reply:
I'm investigating a bit further, and it's almost as if the reactivity of `href` is lost when nested.